OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py

```diff
@@ -152,6 +169,93 @@ def setParameterNode(self, inputParameterNode: Optional[OpenLIFUSonicationPlanne
+
+    @vtk.calldata_type(vtk.VTK_OBJECT)
+    def onNodeRemoved(self, caller, event, node : slicer.vtkMRMLNode) -> None:
+        """ Update volume and target combo boxes when nodes are added to the scene"""
+        self.updateComboBoxOptions()
+
+    @vtk.calldata_type(vtk.VTK_OBJECT)
+    def onNodeAdded(self, caller, event, node : slicer.vtkMRMLNode) -> None:
+        """ Update volume and target combo boxes when nodes are removed from the scene"""
+        self.updateComboBoxOptions()
+
+    def updateComboBoxOptions(self):
+        """" Update all the combo boxes based on the openLIFU objects
+        loaded into the scene. The protocol and transducer information is stored in the openLIFUDataModule's parameter node.
+        The volumes and fiducials are tracked as vtkMRML nodes. 
+        The dropdowns are disabled if the relevant OpenLIFU objects/nodes aren't found"""
+
+        dataLogicParameterNode = self.OpenLIFUDataLogic.getParameterNode()
+
+        # Update parameter combo box
+        self.ui.ProtocolComboBox.clear() 
+        if len(dataLogicParameterNode.loaded_protocols) == 0:
+            self.ui.ProtocolComboBox.addItems(["Select a Protocol"])
+            self.ui.ProtocolComboBox.setDisabled(True)
+        else:
+            self.ui.ProtocolComboBox.setEnabled(True)
+            for protocol in dataLogicParameterNode.loaded_protocols.values():
+                self.ui.ProtocolComboBox.addItem("{} (ID: {})".format(protocol.protocol.name,protocol.protocol.id), protocol) 
+    
+        # Update transducer combo box
+        self.ui.TransducerComboBox.clear()
+        if len(dataLogicParameterNode.loaded_transducers) == 0:
+            self.ui.TransducerComboBox.addItems(["Select a Transducer"]) 
+            self.ui.TransducerComboBox.setDisabled(True)
+        else:
+            self.ui.TransducerComboBox.setEnabled(True)
+            for transducer in dataLogicParameterNode.loaded_transducers.values():
+                transducer_openlifu = transducer.transducer.transducer
+                self.ui.TransducerComboBox.addItem("{} (ID: {})".format(transducer_openlifu.name,transducer_openlifu.id), transducer) 
+
+        # Update volume combo box 
+        self.ui.VolumeComboBox.clear()  
+        if len(slicer.util.getNodesByClass('vtkMRMLScalarVolumeNode')) == 0:
+            self.ui.VolumeComboBox.addItems(["Select a Volume"])
+            self.ui.VolumeComboBox.setDisabled(True)
+        else:
+            self.ui.VolumeComboBox.setEnabled(True)
+            for volume_node in slicer.util.getNodesByClass('vtkMRMLScalarVolumeNode'):
+                self.ui.VolumeComboBox.addItem("{} (ID: {})".format(volume_node.GetName(),volume_node.GetID()), volume_node)
+          
+        # Update target combo box 
+        self.ui.TargetComboBox.clear()  
+        if len(slicer.util.getNodesByClass('vtkMRMLMarkupsFiducialNode')) == 0:
+            self.ui.TargetComboBox.addItem(["Select a Target"])
```

Comment:
this currently shows up as a string represetnation of the list
![image](https://github.com/user-attachments/assets/ef72dd40-c9ca-45e4-898a-bafab92c6a6b)

Reply:
fixed!